src/dotnet-svcutil/lib/src/CodeDomFixup/CodeDomVisitors/AddAsyncOpenClose.cs

```diff
@@ -7,52 +7,43 @@
 using System.ServiceModel;
 using System.Threading.Tasks;
 using System.Linq;
-using System.Xml.Linq;
 
 namespace Microsoft.Tools.ServiceModel.Svcutil
 {
     internal class AddAsyncOpenClose : ClientClassVisitor
     {
+        //default behavior: won't generate CloseAsync()
         private bool _generateCloseAsync = false;
         private bool _addCondition = false;
 
         public AddAsyncOpenClose(CommandProcessorOptions options)
         {
-            if (options.Project != null && options.Project.TargetFrameworks.Count() > 1 && options.Project.TargetFrameworks.Any(t => TargetFrameworkHelper.IsSupportedFramework(t, out FrameworkInfo netfxInfo) && !netfxInfo.IsDnx))
+            if (options.Project != null)
             {
-                _generateCloseAsync = true;
-                FrameworkInfo dnxInfo = null;
-                var tfx = options.Project.TargetFrameworks.FirstOrDefault(t => TargetFrameworkHelper.IsSupportedFramework(t, out dnxInfo) && dnxInfo.IsDnx);
-                if (!string.IsNullOrEmpty(tfx) && dnxInfo.Version.Major >= 6)
-                {
-                    _addCondition = true;
-                }
-            }
-            else
-            {
-                if (options.TargetFramework.IsDnx)
+                Version lowestNetcoreVer = TargetFrameworkHelper.GetLowestNetCoreVersion(options.Project.TargetFrameworks);
+                bool containsNet6OrGreater = options.Project.TargetFrameworks.Any(t => TargetFrameworkHelper.IsSupportedFramework(t, out FrameworkInfo netfxInfo) && netfxInfo.IsDnx && netfxInfo.Version.Major >= 6);
+                bool containsNetFx = options.Project.TargetFrameworks.Any(t => TargetFrameworkHelper.IsSupportedFramework(t, out FrameworkInfo netfxInfo) && !netfxInfo.IsDnx);
+
+                // netfx or lowest.net core < 6.0
+                if (lowestNetcoreVer == null || (lowestNetcoreVer != null && lowestNetcoreVer.Major < 6))
                 {
-                    if (TargetFrameworkHelper.NetCoreVersionReferenceTable.TryGetValue(options.TargetFramework.Version, out var referenceTable))
+                    _generateCloseAsync = true;
+                    if (containsNet6OrGreater)
                     {
-                        string version = referenceTable.FirstOrDefault().Version;
-                        string[] vers = version.Split('.');
-                        if (vers.Length > 1)
-                        {
-                            Version v = new Version(int.Parse(vers[0]), int.TryParse(vers[1], out int minor) ? minor : 0);
-                            // For .NETCore targetframework found in the referenced table, generate CloseAsync() when WCF package version is less than 4.10
-                            if (v.CompareTo(new Version(4, 10)) < 0)
-                            {
-                                _generateCloseAsync = true;
-                            }
-                        }
+                        _addCondition = true;
```

Comment:
It sees that _addCondition and _generateCloseAsync are duplicate?  Could they be merged into one variable?

Reply:
The variable _generateCloseAsync controls whether the CloseAsync() code should be generated, the variable _addCondition determines whether a condition should be added to the generated CloseAsync method in scenarios involving multiple target frameworks. These variables serve different purposes and are not duplicates.